compiler/src/dotty/tools/dotc/Compiler.scala

```diff
@@ -52,6 +52,7 @@ class Compiler {
     List(new FirstTransform,         // Some transformations to put trees into a canonical form
          new CheckReentrant,         // Internal use only: Check that compiled program has no data races involving global vars
          new ElimPackagePrefixes,    // Eliminate references to package prefixes in Select nodes
+         new SetRootTree,            // Set the `rootTreeOrProvider` on class symbols
```

Comment:
When unpickling a symbol, the `rootTreeOrProvider` contains the tree with as it was pickled. Here you are registering the tree after transformations have happened. This may lead us into trouble because `rootTreeOrProvider` will return different trees depending of how we got it.

Reply:
It would make more seance to have the trees that are after PostTyper has transformed them.